### ISSUE TITLE
Query : Missing null propagation logic for some complex queries with manually created GroupJoin-SelectMany-DefaultIfEmpty

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#7761]
Query:

```
from l1_outer in ctx.LevelOne
join subquery in
	(
		from l2_inner in ctx.LevelTwo
		join l1_inner in ctx.LevelOne on l2_inner.Level1_Required_Id equals l1_inner.Id
		select l2_inner
	)
on l1_outer.Id equals subquery.Level1_Optional_Id into grouping
from subquery in grouping.DefaultIfEmpty()
select (int?)subquery.Id;
```

problem is that in final projection subquery can be null, however do don't add null protection logic to this particular call

**Comment by maumar** [comment]
another case:

```cs
from l1 in ctx.LevelOne
from l2 in l1.OneToMany_Optional.Where(l => l.Id > 5).OrderByDescending(l => l.Name).DefaultIfEmpty()
select l1.Name + " " + l2.Name
```

**Comment by maumar** [comment]
Fixed in bb7eebde422e6bf9601f4c97c72492dd52f0fe41
